tests/unit/api/metadataSearch.js

```diff
@@ -0,0 +1,258 @@
+const assert = require('assert');
+const querystring = require('querystring');
+
+const async = require('async');
+const { parseString } = require('xml2js');
+
+const metadataSearch = require('../../../lib/api/metadataSearch');
+const { bucketPut } = require('../../../lib/api/bucketPut');
+const objectPut = require('../../../lib/api/objectPut');
+const { cleanup, DummyRequestLogger, makeAuthInfo } = require('../helpers');
+const DummyRequest = require('../DummyRequest');
+
+const { errors } = require('arsenal');
+
+const authInfo = makeAuthInfo('accessKey1');
+const bucketName = 'bucketname';
+const delimiter = '/';
+const log = new DummyRequestLogger();
+const namespace = 'default';
+const postBody = Buffer.from('I am a body', 'utf8');
+const prefix = 'sub';
+
+const objectName1 = `${prefix}${delimiter}objectName1`;
+const objectName2 = `${prefix}${delimiter}objectName2`;
+const objectName3 = 'invalidURI~~~b';
+const objectName4 = `${objectName1}&><"\'`;
+const testPutBucketRequest = new DummyRequest({
+    bucketName,
+    headers: {},
+    url: `/${bucketName}`,
+    namespace,
+}, Buffer.alloc(0));
+const testPutObjectRequest1 = new DummyRequest({
+    bucketName,
+    headers: { 'x-amz-meta-test': 'test' },
+    url: `/${bucketName}/${objectName1}`,
+    namespace,
+    objectKey: objectName1,
+}, postBody);
+const testPutObjectRequest2 = new DummyRequest({
+    bucketName,
+    headers: { 'x-amz-meta-test': 'test' },
+    url: `/${bucketName}/${objectName2}`,
+    namespace,
+    objectKey: objectName2,
+}, postBody);
+const testPutObjectRequest3 = new DummyRequest({
+    bucketName,
+    headers: {},
+    url: `/${bucketName}/${objectName3}`,
+    namespace,
+    objectKey: objectName3,
+}, postBody);
+const testPutObjectRequest4 = new DummyRequest({
+    bucketName,
+    headers: {},
+    url: `/${bucketName}/${objectName3}`,
+    namespace,
+    objectKey: objectName4,
+}, postBody);
+
+const baseGetRequest = {
+    bucketName,
+    namespace,
+    headers: { host: '/' },
+};
+const baseUrl = `/${bucketName}`;
+
+const tests = [
```

Comment:
So, we are not testing the "AccessDenied" case (i.e., whether Vault successfully handle this new API and permissions or not) here, right?

Reply:
Also, what about the `tests/functional/aws-node-sdk/test/mdSearch/basicSearch.js` file that implements some md tests? Is it related to this API?

Reply:
> So, we are not testing the "AccessDenied" case (i.e., whether Vault successfully handle this new API and permissions or not) here, right?

I think no, interacting with Vault happens outside of this metadataSearch() function

Reply:
Thanks!